org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/InvisibleProjectImporter.java

```diff
@@ -144,20 +154,20 @@ public static boolean loadInvisibleProject(IPath javaFile, IPath rootPath, boole
 		if (preferencesManager != null && preferencesManager.getPreferences() != null) {
 			sourcePathsFromPreferences = preferencesManager.getPreferences().getInvisibleProjectSourcePaths();
 		}
-		List<IPath> sourcePaths;
+		Set<IPath> sourcePaths = new HashSet<>();
 		if (sourcePathsFromPreferences != null) {
```

Comment:
Yes, I assume that if user defines that setting, they are clear about the project structure they want to have. Then we skip the inference to avoid any unexpected behaviors.